sdk/javascript/src/symbol/models_ts.js

```diff
@@ -0,0 +1,1039 @@
+	 * @param {Hash256} parentHash Hash of the AggregateBondedTransaction that is signed by this cosignature.
+	 */
+	constructor(version, signerPublicKey, signature, parentHash) {
+		this.rawDescriptor = {
+			version,
+			signerPublicKey,
+			signature,
+			parentHash
+		};
+	}
+
+	/**
+	 * Builds a representation of this descriptor that can be passed to a factory function.
+	 * @returns {object} Descriptor that can be passed to a factory function.
+	 */
+	toMap() {
+		return this.rawDescriptor;
+	}
+}
+/**
+ * Type safe descriptor used to generate a descriptor map for AggregateCompleteTransactionV1Descriptor.
+ *
+ * Send transactions in batches to different accounts (V1, deprecated).
+ * Use this transaction when all required signatures are available when the transaction is created.
+ */
+export class AggregateCompleteTransactionV1Descriptor {
+	/**
+	 * Creates a descriptor for AggregateCompleteTransactionV1.
+	 * @param {Hash256} transactionsHash Hash of the aggregate's transaction.
+	 * @param {models.EmbeddedTransaction[]|undefined} transactions Embedded transaction data.
+	Transactions are variable-sized and the total payload size is in bytes.
+	Embedded transactions cannot be aggregates.
+	 * @param {models.Cosignature[]|undefined} cosignatures Cosignatures data.
+	Fills up remaining body space after transactions.
+	 */
+	constructor(transactionsHash, transactions = undefined, cosignatures = undefined) {
+		this.rawDescriptor = {
+			type: 'aggregate_complete_transaction_v1',
+			transactionsHash
+		};
+
+		if (transactions)
+			this.rawDescriptor.transactions = transactions;
+
+		if (cosignatures)
+			this.rawDescriptor.cosignatures = cosignatures;
+	}
+
+	/**
+	 * Builds a representation of this descriptor that can be passed to a factory function.
+	 * @returns {object} Descriptor that can be passed to a factory function.
+	 */
+	toMap() {
+		return this.rawDescriptor;
+	}
+}
+/**
+ * Type safe descriptor used to generate a descriptor map for AggregateCompleteTransactionV2Descriptor.
+ *
+ * Send transactions in batches to different accounts (V2, latest).
+ * Use this transaction when all required signatures are available when the transaction is created.
+ */
+export class AggregateCompleteTransactionV2Descriptor {
+	/**
+	 * Creates a descriptor for AggregateCompleteTransactionV2.
+	 * @param {Hash256} transactionsHash Hash of the aggregate's transaction.
+	 * @param {models.EmbeddedTransaction[]|undefined} transactions Embedded transaction data.
+	Transactions are variable-sized and the total payload size is in bytes.
+	Embedded transactions cannot be aggregates.
+	 * @param {models.Cosignature[]|undefined} cosignatures Cosignatures data.
+	Fills up remaining body space after transactions.
+	 */
+	constructor(transactionsHash, transactions = undefined, cosignatures = undefined) {
+		this.rawDescriptor = {
+			type: 'aggregate_complete_transaction_v2',
+			transactionsHash
+		};
+
+		if (transactions)
```

Comment:
What is the use case when there is no transaction in the aggregate tx?

Reply:
there is none, transaction will get rejected with `Failure_Aggregate_No_Transactions` by network 

the generator is merely flagging all arrays and conditional arguments as optional